packages/core/src/ceramic.ts

```diff
@@ -208,11 +206,11 @@ class Ceramic implements CeramicApi {
 
     this._doctypeHandlers = new HandlersMap(this._logger)
 
-    this._repository = modules.repository
-    this._repository.setStateStore(this.pinStore.stateStore)
-    const documentFactory = new DocumentFactory(this.dispatcher, this.pinStore, this.context, this._validateDocs, this._doctypeHandlers)
-    this._repository.setDocumentFactory(documentFactory)
-    this.loadingQueue = new LoadingQueue(this._repository, this.dispatcher, this._doctypeHandlers, this.context, this.pinStore, this._logger, documentFactory)
+    const documentFactory = new DocumentFactory(this.dispatcher, pinStore, this.context, this._validateDocs, this._doctypeHandlers)
```

Comment:
Ideally we'd eventually be in the world where literally nothing besides the `Repository` knows about the `pinStore`.  Looks like `DocumentFactory` and `Document` both still access `pinStore` directly, but it seems like both of those are going away or getting majorly re-worked before this project is done, so maybe their use of `pinStore` will go away naturally in a future commit?

Reply:
Almost every state change has to be stored as `Document#_updateStateIfPinned`, so no way _for now_ this goes away. One could get rid of it at a later stage, after we have a running state.